python/lsst/ts/standardscripts/auxtel/latiss.py

```diff
@@ -0,0 +1,244 @@
+            await self.take_image(exptime=exptime, shutter=True, image_seq_name=tag,
+                                  latiss_filter=latiss_filter, latiss_grating=latiss_grating,
+                                  latiss_linear_stage=latiss_linear_stage)
+
+    async def take_image(self, exptime, shutter, image_seq_name,
+                         latiss_filter=None, latiss_grating=None, latiss_linear_stage=None,
+                         science=True, guide=False, wfs=False,
+                         ):
+        """Encapsulates the action of setting up the spectrograph and taking
+        images.
+
+        Setting up the spectrograph and taking images cannot be done
+        concurrently. One needs first to setup the spectrograph then,
+        request images.
+
+        Parameters
+        ----------
+        exptime : `float`
+            The exposure time for the image, in seconds.
+        shutter : `bool`
+            Should activate the shutter? (False for bias and dark)
+        image_seq_name : `str`
+            A string to identify the image.
+        latiss_filter : `None` or `int` or `str`
+            Filter id or name. If None, ignore setting filter.
+        latiss_grating : `None` or `int` or `str`
+            Grating id or name.  If None, ignore setting grating.
+        latiss_linear_stage : `None` or `float`
+            Linear stage position.  If None, ignore setting linear stage.
+        science : `bool`
+            Mark image as science (default=True)?
+        guide : `bool`
+            Mark image as guide (default=False)?
+        wfs : `bool`
+            Mark image as wfs (default=False)?
+
+        Returns
+        -------
+        endReadout : `SALPY_ATCamera.ATCamera_logevent_endReadoutC`
+
+        """
+
+        await self.setup_atspec(latiss_filter=latiss_filter,
+                                latiss_grating=latiss_grating,
+                                latiss_linear_stage=latiss_linear_stage)
+
+        return await self.expose(exp_time=exptime, shutter=shutter, image_seq_name=image_seq_name,
+                                 science=science, guide=guide, wfs=wfs)
+
+    async def expose(self, exp_time, shutter, image_seq_name,
+                     science=True, guide=False, wfs=False):
+        """Encapsulates the take image command.
+
+        This basically consists of configuring and sending a takeImages
+        command to the camera and waiting for an endReadout event.
+
+        Parameters
+        ----------
+        exp_time : `float`
+            The exposure time for the image, in seconds.
+        shutter : `bool`
+            Should activate the shutter? (False for bias and dark)
+        image_seq_name : `str`
+            A string to identify the image.
+        science : `bool`
+            Mark image as science (default=True)?
+        guide : `bool`
+            Mark image as guide (default=False)?
+        wfs : `bool`
+            Mark image as wfs (default=False)?
```

Comment:
Consider using an enum class instead of all these flags. It is simple and it needs to changes to the argument list if another image type is added.

Assuming "science", "guide" and "wfs" are mutually exclusive you can use enum.Enum, if they can mix then enum.Flag.

    class ImageType(enum.Enum):
        SCIENCE = enum.auto()
        GUIDE = enum.auto()
        WAVEFRONT_SENSOR = enum.auto()
        OTHER = enum.auto()  # if this is a valid option

    image_type : `ImageType`
        Image type

then in your code:

    image_type = ImageType(image_type)  # to cast integers to enums and reject invalid values
    if image_type == ImageType.SCIENCE...

Even if they can be used together there are enums class for that

Reply:
I'm not entirely sure how these are used in the camera. I'll investigate it thought. I liked your suggestion but I'm not sure I'll be able to implement it in this ticket.